yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -359,4 +362,31 @@ export class AztecNodeService implements AztecNode {
       globalsHash,
     );
   }
+
+  /**
+   * Returns an instance of MerkleTreeOperations having first ensured the world state is fully synched
+   * @returns An instance of a committed MerkleTreeOperations
+   */
+  private async getWorldState() {
+    await this.syncWorldState();
+    return this.worldStateSynchroniser.getCommitted();
+  }
+
+  /**
+   * Ensure we fully sync the world state
+   * @returns A promise that fulfils once the world state is synced
+   */
+  private async syncWorldState() {
+    while (true) {
+      const [blockSourceHeight, worldStateBlock] = await Promise.all([
+        this.blockSource.getBlockHeight(),
+        this.worldStateSynchroniser.status().then(x => x.syncedToL2Block),
+      ]);
```

Comment:
Do we need to keep updating the "target" `blockSourceHeight` on every iteration? Can't we just pin the current `blockSourceHeight` and loop until we reach it, so we avoid constantly running behind it?

If so, maybe it makes sense to change the `syncImmediate` method to something like `syncImmediateToAtLeastThisBlockNumber`. This would also make it more consistent with the `wait`ing method we implemented in aztec.js, where we don't force the RPC to be up to date with whatever's latest, but with the block where the target tx was mined (see [here](https://github.com/AztecProtocol/aztec-packages/blob/89e4e1ac5e90905aa475ba2f8b6afb7b77dc772a/yarn-project/aztec.js/src/contract/sent_tx.ts#L96-L98)).

Reply:
I get the sense we'll need to review these APIs and add a block tag to most methods, like Ethereum has. Or a "synced at least until this point" block tag.

Reply:
Updated the `syncImmediate` call to take an optional `blockHeight` argument representing the minimum block that should be reached.